src/System.Management.Automation/engine/GetCommandCommand.cs

```diff
@@ -1694,40 +1694,66 @@ private static PSObject GetParameterType(Type parameterType)
     }
 
     /// <summary>
+    /// Provides argument completion for Noun parameter. 
     /// </summary>
     public class NounArgumentCompleter : IArgumentCompleter
     {
+        /// </summary>
+        /// Returns completion results for Noun parameter.
+        /// </summary>
+        /// <param name="commandName">The command name.</param>
+        /// <param name="parameterName">The parameter name.</param>
+        /// <param name="wordToComplete">The word to complete.</param>
+        /// <param name="commandAst">The command AST.</param>
+        /// <param name="fakeBoundParameters">The fake bound parameters.</param>
+        /// <returns>List of completion results.</returns>
+        public IEnumerable<CompletionResult> CompleteArgument(
+            string commandName,
+            string parameterName,
+            string wordToComplete,
+            CommandAst commandAst,
+            IDictionary fakeBoundParameters) => CompletionCompleters.GetMatchingResults(
+                wordToComplete,
+                possibleCompletionValues: GetCommandNouns(fakeBoundParameters).Order());
```

Comment:
I do wonder if the overhead from `Order()` for LINQ is necessary. I brought this over from the previous code.

Probably too late to change that now 😄.

Reply:
Should be resolved now with https://github.com/PowerShell/PowerShell/pull/24977#discussion_r1948102230